File: devjobs-web-api/Entities/Job.cs
using System.Text.Json.Serialization;

namespace devjobs_web_api.Entities;

public class Job
{
    public int Id { get; set; }
    public Company Company { get; set; }
    public int CompanyId { get; set; }
    public string Position { get; set; }
    public DateTime PostedAt { get; set; }
    public string ContractType { get; set; }
    public string Location { get; set; }
    public string Description { get; set; }
    public Requirements Requirements { get; set; }
    public int RequirementsId { get; set; }
    public Role Role { get; set; }
    public int RoleId { get; set; }
}

File: devjobs-web-app/src/app/@types/Job.ts
import { Company, Contract } from "./index"

export interface Job {
    id: number,
    company: Company,
    companyId: number,
    position: string,
    postedAt: Date,
    contract: Contract,
    location: string,
    description: string,
    requirements: Requirements,
    requirementsId: number,
    role: Role,
    roleId: number
}

export interface JobSummary {
    id: number,
    company: string,
    postedAt: Date,
    position: string,
    contract: Contract,
    location: string,
    backgroundColor: string
}

interface Requirements {
    id: number,
    jobId: number,
    content: string,
    items: [
        {
            id: number,
            requirementsId: number,
            description: string
        }
    ]
}

interface Role {
    id: number,
    jobId: number,
    content: string,
    items: [
        {
            id: number,
            roleId: number,
            description: string
        }
    ]
}